odoo/addons/base/tests/test_db_cursor.py

```diff
@@ -127,8 +128,8 @@ class TestTestCursor(common.TransactionCase):
     def setUp(self):
         super().setUp()
         # make the registry in test mode
-        self.registry.enter_test_mode(self.cr)
-        self.addCleanup(self.registry.leave_test_mode)
+        self.registry_enter_test_mode()
```

Comment:
Just an idea, we could avoid the systematic addCleanup since registry_enter_test_mode is now on the test class, it could be automatic. 

We just need to be careful that if called in a setupClass (so called on the class), it would be an addClassCleanup.

Reply:
Can we differentiate it though? 🤔

Reply:
The instance will need to be provided through params on the method, there is little need for individual test based registry mode without using the already available context manager.

Reply:
I'm not sure. 
Technically yes, we could so something like `sys._getframe(1).f_code.co_name == 'setUp'`

Another solution would be to do as the current way to start class and test patcher, have different methods

cls.class_enter_test_mode
self.enter_test_mode

We could also remove the class version an only allow that in the setup.

Reply:
What about the current solution?